src/main/java/org/osc/controller/nuage/api/NuageSecurityControllerApi.java

```diff
@@ -249,6 +252,41 @@ public Element registerRedirectionTarget(InspectionPortElement inspectionPort, S
         List<Domain> dms = fetcher.fetch(filter, null, null, null, null, null, Boolean.FALSE);
         if (!CollectionUtils.isEmpty(dms)) {
             selectDomain = dms.get(0);
+
+            List<NetworkElement> inspectionPorts = new ArrayList<>();
```

Comment:
Could you add all the new code in a separate private method in order to keep this one a bit smaller? Also on the new method if you could use a clear name and javadoc highlighting why this is it needed it would be great: The loop happens because of this and that and this method breaks it by adding.... etc.  Same thing for the deletion.